src/Components/Input.tsx

```diff
@@ -12,276 +10,57 @@ export interface InputProps extends React.HTMLProps<HTMLInputElement> {
-      this.props.onChange(e)
-    }
-  }
-
-  getRightViewForPassword() {
-    const icon = this.state.showPassword ? (
-      <ClosedEyeIcon onClick={this.toggleShowPassword} />
-    ) : (
-      <OpenEyeIcon onClick={this.toggleShowPassword} />
-    )
-
-    return <Eye onClick={this.toggleShowPassword}>{icon}</Eye>
-  }
-
-  toggleShowPassword = () => {
-    this.setState({
-      showPassword: !this.state.showPassword,
-    })
-  }
-
-  get convertedType() {
-    const { type } = this.props
-    if (this.state.showPassword && type === "password") {
-      return "text"
-    }
-    return type
-  }
-
-  render() {
-    const { error, quick } = this.props
-
-    if (quick) {
-      // prettier-ignore
-      const {
-        className,
-        label,
-        leftView,
-        ref: _ref,
-        rightView,
-        showPasswordMessage,
-        type,
-        onChange,
-        setTouched,
-        ...newProps
-      } = this.props
-      const showLabel = (!!this.state.focused || !!this.state.value) && !!label
-      const isPassword = type === "password"
-
-      return (
-        <Container>
-          <InputContainer
-            hasLabel={!!label}
-            hasError={!!error}
-            className={this.state.focused ? "focused" : ""}
-          >
-            <Label out={!showLabel}>{label}</Label>
-            {!!leftView && leftView}
-            <InputComponent
-              innerRef={this.props.innerRef}
-              {...newProps}
-              onFocus={this.onFocus}
-              onBlur={this.onBlur}
-              onChange={this.onChange}
-              value={this.state.value}
-              type={this.convertedType}
-              showLabel={showLabel}
-            />
-            {isPassword
-              ? this.getRightViewForPassword()
-              : !!rightView && rightView}
-          </InputContainer>
-          {!error && showPasswordMessage ? (
-            <PasswordMessage>
-              Password must be at least 8 characters.
-            </PasswordMessage>
-          ) : (
-            ""
-          )}
-          <Error show={!!error}>{error}</Error>
-        </Container>
-      )
-    }
-
-    const { title, description, ref, ...rest } = this.props
-    return (
-      <Container>
-        {title && <Title>{title}</Title>}
-        {description && <Description>{description}</Description>}
-        <StyledInput hasError={!!error} {...rest} />
-        <Error show={!!error}>{error}</Error>
-      </Container>
-    )
-  }
+export const Input: SFC<InputProps> = ({
+  error,
+  title,
+  description,
+  ref,
+  ...rest
+}) => {
+  return (
+    <Container>
+      {title && <Title>{title}</Title>}
+      {description && <Description>{description}</Description>}
+      <StyledInput hasError={!!error} {...rest} />
+      {error && <InputError>{error}</InputError>}
+    </Container>
+  )
 }
 
 const Container = styled.div`
-  padding-bottom: 5px;
+  padding-bottom: ${space(0.5)}px;
```

Comment:
Totally cool with ^ but thinking that it might be a little more future-friendly if this was refactored so that it behaved in a more palette-like way -- meaning: 

```jsx
<Box pb={0.5}>
  ...
</Box>
```

Then we can lose the intermediate variable and some indirection.

Reply:
Yeah, we talked about that. We thought those types of changes would fit better with the work in palette - the goal here being to properly identify the seams so that we can dig the inputs out once they're moved, plus a little bit of "easy" palette-izing. 

If you feel strongly that this work should happen here in reaction, we can work on that next week.

Reply:
Not  strongly! This is totally fine 👍